cmd/cosign/cli/attest.go

```diff
@@ -80,6 +80,7 @@ func Attest() *cobra.Command {
 				OIDCProvider:             o.OIDC.Provider,
 				SkipConfirmation:         o.SkipConfirmation,
 				TSAServerURL:             o.TSAServerURL,
+				TlogUpload:               o.TlogUpload,
```

Comment:
It worries me that this flag is duped between KeyOpts and AttestCommand. Instead of adding it to KeyOpts, can we print the skip confirmation message sometime during the flow the TLOG upload is actually happening (instead of being inside a signer)?

A pretty clean place to put it would probably be `ShouldUploadToTlog` -- we already check for image being public there, so makes sense to put the human confirmation. If the person denies the confirmation return false.

It probably addresses Hayden's issue of using your own key and tlog upload all in one go

Reply:
SGTM, updated!